### PR TITLE
Create MAIN timer

### PR DESCRIPTION
**Pull request by MykhailenkoAnton**
Currently, the timer works how it should work. Added what would when the user press on rest_work timer is reset. Added update ClockFace label every frame. Added stop or start it with a play/pause button.

### CODE REVIEW

File: gd_cpp_lib/src/MainNode.cpp [hunk 1]
```diff
@@ -51,18 +52,123 @@ void MainNode::_ready()
 
     _playPauseButton =
         cast_to<godot::TextureButton>(_GUI->find_child(godot::String(node_names::PLAY_PAUSE_BUTTON_NAME)));
+
+    _toggleButtonWorkRest =
+        cast_to<godot::CheckButton>(_GUI->find_child(godot::String(node_names::TOGGLE_BUTTON_WORK_REST_NAME)));
+
+    _WorkTextLabel = cast_to<godot::Label>(_GUI->find_child(godot::String(node_names::WORK_TEXT_LABEL_NAME)));
+    _RestTextLabel = cast_to<godot::Label>(_GUI->find_child(godot::String(node_names::REST_TEXT_LABEL_NAME)));
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.cpp:60]: _WorkTextLabel  -> _**w**orkTextLabel 
_RestTextLabel -> _**r**estTextLabel 
please

File: gd_cpp_lib/src/MainNode.cpp [hunk 2]
```diff
@@ -51,18 +52,123 @@ void MainNode::_ready()
 
     _playPauseButton =
         cast_to<godot::TextureButton>(_GUI->find_child(godot::String(node_names::PLAY_PAUSE_BUTTON_NAME)));
+
+    _toggleButtonWorkRest =
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.cpp:56]: _toggleButtonWorkRest -> _workRestToggleButton
**name**Type at the end, for example line above _**playPause**Button 
or below _**workText**Label

File: gd_cpp_lib/src/MainNode.h [hunk 3]
```diff
@@ -36,8 +47,15 @@ class MainNode : public godot::Node
...
     godot::Timer* _timer = nullptr;
     godot::TextureButton* _playPauseButton = nullptr;
+    godot::CheckButton* _toggleButtonWorkRest = nullptr;
+    godot::Label* _WorkTextLabel = nullptr;
+    godot::Label* _RestTextLabel = nullptr;
 
     eTimerState _timerState = eTimerState::PAUSE;
+    eTimerMode _timerMode = eTimerMode::WORK;
+
+    int _allTimeWorkInMin = 25;
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.h:57]: not clear name for me, try something like **_workingTimeInMin**

File: gd_cpp_lib/src/MainNode.h [hunk 4]
```diff
@@ -36,8 +47,15 @@ class MainNode : public godot::Node
...
+    int _allTimeRestInMin = 5;
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.h:58]: _restTimeInMin

File: gd_cpp_lib/src/MainNode.h [hunk 5]
```diff
@@ -22,11 +29,15 @@ class MainNode : public godot::Node
...
+    void _process(double delta) override;
 
+    double getTimeInSec();
     void _on_play_pause_button_toggled(bool bIsPlayed);
-
-    void _ready() override;
+    void _on_work_rest_button_toggled(bool bIsWork);
+    void showStartTimeOnScreen(int time);
+    void setColorTextForTimerMode(eTimerMode timerMode);
+    void updateTimerTextOnScene();
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.h:40]: can be private

File: gd_cpp_lib/src/MainNode.h [hunk 6]
```diff
@@ -22,11 +29,15 @@ class MainNode : public godot::Node
     MainNode();
     ~MainNode();
 
-    void _on_button_pressed();
+    void _ready() override;
+    void _process(double delta) override;
 
+    double getTimeInSec();
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.h:35]: can be private

File: gd_cpp_lib/src/MainNode.cpp [hunk 7]
```diff
@@ -51,18 +52,123 @@ void MainNode::_ready()
...
+    else if (timerMode == eTimerMode::REST)
+    {
+        _WorkTextLabel->add_theme_color_override("font_color", godot::Color(1, 1, 1, 1));
+        _RestTextLabel->add_theme_color_override("font_color", godot::Color(0.3, 1, 1, 1));
+    }
+}
+
+void MainNode::updateTimerTextOnScene()
+{
+    double timeLeft = _timer->get_time_left();
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.cpp:164]: can be const

File: gd_cpp_lib/src/MainNode.cpp [hunk 8]
```diff
@@ -51,18 +52,123 @@ void MainNode::_ready()
...
+
+    int minTime = timeLeft / SEC_IN_MIN;
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.cpp:166]: const int

File: gd_cpp_lib/src/MainNode.cpp [hunk 9]
```diff
@@ -51,18 +52,123 @@ void MainNode::_ready()
...
+    int secTime = (static_cast<int>(timeLeft)) % SEC_IN_MIN;
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.cpp:167]: (static_cast<int>(timeLeft)) -> static_cast<int>(timeLeft)
no need extra bracer here

File: gd_cpp_lib/src/MainNode.cpp [hunk 10]
```diff
@@ -51,18 +52,123 @@ void MainNode::_ready()
...
+
+    godot::String timeInMin = godot::String::num_int64(minTime);
+    godot::String timeInSec = godot::String::num_int64(secTime);
```
Comment (ydementieiev) [inline gd_cpp_lib/src/MainNode.cpp:170]: please add 0 for min also

**Comment by MykhailenkoAnton** [comment]
All done. Please check the logic with the addition of "0".

**Review by ydementieiev — APPROVED** [review]

**Review by ydementieiev — APPROVED** [review]

**Pull request merged**